openfe/tests/utils/test_system_probe.py

```diff
@@ -0,0 +1,319 @@
+import contextlib
+from collections import namedtuple
+import sys
+from unittest.mock import Mock, patch
+
+import psutil
+import pytest
+
+from openfe.utils.system_probe import (
+    _get_disk_usage,
+    _get_gpu_info,
+    _get_hostname,
+    _get_psutil_info,
+    _probe_system,
+)
+
+
+# Named tuples from https://github.com/giampaolo/psutil/blob/master/psutil/_pslinux.py
+svmem = namedtuple(
+    "svmem",
+    [
+        "total",
+        "available",
+        "percent",
+        "used",
+        "free",
+        "active",
+        "inactive",
+        "buffers",
+        "cached",
+        "shared",
+        "slab",
+    ],
+)
+pfullmem = namedtuple(
+    "pfullmem",
+    ["rss", "vms", "shared", "text", "lib", "data", "dirty", "uss", "pss", "swap"],
+)
+
+
+@contextlib.contextmanager
+def patch_system():
+    # single patch to fully patch the system
+    patch_hostname = patch("socket.gethostname", Mock(return_value="mock-hostname"))
+
+    patch_psutil_Process_as_dict = patch(
+        "psutil.Process.as_dict",
+        Mock(
+            return_value={
+                "pid": 1590579,
+                "status": "running",
+                "exe": "/home/winry/micromamba/envs/openfe/bin/python3.10",
```

Comment:
I was expecting `/home/mmh/` here... who's this `winry`? 🤔

Reply:
hehe, I cleaned it out and thought I would put a reference to something :smile: